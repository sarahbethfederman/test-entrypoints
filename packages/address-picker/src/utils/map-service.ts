export default class MapService {
  private placesService: google.maps.places.PlacesService = {} as google.maps.places.PlacesService;
  private autocompleteService: google.maps.places.AutocompleteService = new google.maps.places.AutocompleteService();
  private sessionToken: google.maps.places.AutocompleteSessionToken = new google.maps.places.AutocompleteSessionToken();

  constructor(map?: google.maps.Map) {
    this.placesService = new google.maps.places.PlacesService(map || document.createElement('div'));
  }

  /**
   * Map enum to string
   * @param status google.maps.places.PlacesServiceStatus
   */
  private apiRejectionReason(status: google.maps.places.PlacesServiceStatus): Error {
    if (status === google.maps.places.PlacesServiceStatus.ZERO_RESULTS) {
      return new Error('ZERO_RESULTS');
    } else if (status === google.maps.places.PlacesServiceStatus.INVALID_REQUEST) {
      return new Error('INVALID_REQUEST');
    } else if (status === google.maps.places.PlacesServiceStatus.OVER_QUERY_LIMIT) {
      return new Error('OVER_QUERY_LIMIT');
    } else if (status === google.maps.places.PlacesServiceStatus.NOT_FOUND) {
      return new Error('NOT_FOUND');
    } else if (status === google.maps.places.PlacesServiceStatus.REQUEST_DENIED) {
      return new Error('REQUEST_DENIED');
    } else if (status === google.maps.places.PlacesServiceStatus.UNKNOWN_ERROR) {
      return new Error('UNKNOWN_ERROR');
    }
    return new Error('UNKNOWN_ERROR');
  }

  public placePredictions(
    input: string,
    componentRestrictions: google.maps.places.ComponentRestrictions
  ): Promise<google.maps.places.AutocompletePrediction[]> {
    return new Promise((resolve, reject) => {
      (this.autocompleteService as google.maps.places.AutocompleteService).getPlacePredictions(
        {
          input,
          sessionToken: this.sessionToken,
          componentRestrictions,
        },
        (suggestions: google.maps.places.AutocompletePrediction[], status: google.maps.places.PlacesServiceStatus) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && suggestions !== null) {
            resolve(suggestions);
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }

  public placeDetails(
    placeId: string
  ): Promise<{
    geometry: google.maps.places.PlaceGeometry | undefined;
    addressComponents: google.maps.GeocoderAddressComponent[];
  }> {
    return new Promise((resolve, reject) => {
      (this.placesService as google.maps.places.PlacesService).getDetails(
        {
          placeId,
          sessionToken: this.sessionToken,
        },
        (
          { geometry, address_components = [] }: google.maps.places.PlaceResult,
          status: google.maps.places.PlacesServiceStatus
        ) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && address_components.length > 0) {
            resolve({
              geometry,
              addressComponents: address_components,
            });
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }
}
