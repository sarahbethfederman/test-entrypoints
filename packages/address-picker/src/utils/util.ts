import { AddressObject } from '../components/Modal';
import { STREET_TYPE } from '../components/Modal/constants';
import * as to from 'to-case';

/**
 * Where possible this function takes the users input and
 * Google's address information and enhances it with more
 * accurate subpremise/unit information
 */
export const transformGoogleResponse = (
  label: string,
  res: google.maps.GeocoderAddressComponent[],
  regionSearchOnly?: boolean
): AddressObject => {
  label = label.toLowerCase();

  // pick the easily determined properties from response
  const { short_name: postcode = '' } = res.find((curr) => curr.types.includes('postal_code')) || {};
  let { short_name: state = '' } = res.find((curr) => curr.types.includes('administrative_area_level_1')) || {};
  let { long_name: streetName = '' } = res.find((curr) => curr.types.includes('route')) || {};
  let { short_name: streetNumber = '' } = res.find((curr) => curr.types.includes('street_number')) || {};
  const { long_name: suburb = '' } = res.find((curr) => curr.types.includes('locality')) || {};
  const { long_name: country = '' } = res.find((curr) => curr.types.includes('country')) || {};
  let unit = '';

  // Google has no concept of "street". See if we recognise a streetType from
  // the route then use that to generate streetName and streetType
  // streetType default to 'street' if regionSearchOnly is false
  const street = streetName.trim().split(' ');
  const streetTypeWith = street[street.length - 1];
  let streetType = !regionSearchOnly ? 'Street' : '';
  if (STREET_TYPE.find((street_type) => street_type.label === streetTypeWith)) {
    streetType = streetTypeWith;
  }

  streetName = streetName.replace(streetType, '').trim();

  // edge case, state is listed as JBT for jervis bay when it should be ACT. Naughty Google
  if (state === 'JBT') {
    state = 'ACT';
  }

  // check for unit / subpremise info by crossreferencing against user input
  const streetNumberData = label.substring(0, label.indexOf(streetName.toLowerCase())).trim();
  if (streetNumberData !== streetNumber) {
    if (streetNumberData.includes('/')) {
      unit = streetNumberData.substring(0, streetNumberData.indexOf('/'));
    } else {
      streetNumber = streetNumberData;
    }
  }

  return {
    unit,
    lotSection: '',
    level: '',
    buildingName: '',
    streetNumber,
    streetName,
    streetType,
    suburb,
    postcode,
    state,
    country,
  };
};

export const getFormatedString = (address: AddressObject) => {
  let formatString = address.unit ? ''.concat(address.unit, '/') : '';
  formatString = address.lotSection ? formatString.concat('Lot. ', address.lotSection) : formatString;
  formatString = address.level ? formatString.concat(' level ', address.level) : formatString;
  formatString = address.buildingName ? formatString.concat(' ', to.capital(address.buildingName), ' ') : formatString;
  formatString = address.streetNumber ? formatString.concat(address.streetNumber) : formatString;
  formatString = address.streetName ? formatString.concat(' ', to.capital(address.streetName)) : formatString;
  formatString = address.streetType ? formatString.concat(' ', to.capital(address.streetType)) : formatString;
  formatString = address.suburb ? formatString.concat(', ', to.capital(address.suburb)) : formatString;
  formatString = address.state ? formatString.concat(', ', to.upper(address.state)) : formatString;
  formatString = address.postcode ? formatString.concat(' ', address.postcode) : formatString;
  formatString = address.country ? formatString.concat(', ', to.capital(address.country)) : formatString;
  return formatString;
};
