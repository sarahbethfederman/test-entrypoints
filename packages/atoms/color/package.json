{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  