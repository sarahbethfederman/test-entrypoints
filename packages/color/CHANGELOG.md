2.0.2 - New Changelog
