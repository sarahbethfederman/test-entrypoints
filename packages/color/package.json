{
  "name": "@lendi-ui/color",
  "version": "5.0.0",
  "description": "Color utilities for color.",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^8.0.0",
    "@types/styled-components": "^4.1.18",
    "styled-components": "^4"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^3.0.0"
  },
  "scripts": {
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:cjs"
  }
}
