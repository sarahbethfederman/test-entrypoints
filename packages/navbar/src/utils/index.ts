export const browserSupportsPassiveListeners = () => {
  let supportsPassive = false;
  try {
    var opts = Object.defineProperty({}, 'passive', {
      get: function() {
        supportsPassive = true;
      },
    });
    // for testing browser compatibility, no need to worry
    // about the null TS issue.
    // @ts-ignore
    window.addEventListener('testPassive', null, opts);
    // @ts-ignore
    window.removeEventListener('testPassive', null, opts);
  } catch (e) {}

  return supportsPassive;
};
