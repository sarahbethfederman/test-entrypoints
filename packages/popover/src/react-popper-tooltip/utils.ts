import * as React from 'react';

export const TooltipContext = React.createContext({});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type Fn = (...args: any[]) => void;
type FnOrBool = undefined | Fn;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const callAll = (...fns: FnOrBool[]) => (...args: any[]) => fns.forEach((fn) => fn && fn(...args));

export const noop = () => {};

export const canUseDOM = () => !!(typeof window !== 'undefined' && window.document && window.document.createElement);
