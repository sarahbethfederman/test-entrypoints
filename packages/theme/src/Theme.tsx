import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
