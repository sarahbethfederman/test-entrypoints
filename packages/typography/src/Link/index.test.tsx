import * as React from 'react';
import { shallow, mount } from 'enzyme';

import Theme, { theme } from '@lendi-ui/theme';
import { color as getColor } from '@lendi-ui/color';

import { Link, LinkSize } from '..';
import { deriveSize } from '@lendi-ui/utils';

let wrapper;
const render = (props) => {
  wrapper = mount(
    <Theme>
      <Link {...props} />
    </Theme>
  );
};

describe('Link', () => {
  const sizes: LinkSize[] = ['lg', 'md', 'sm'];
  const colors = ['primary.500', 'shade.0', 'shade.1000'];

  it('should render a <button> when onClick is specified', () => {
    const onClick = jest.fn();
    render({ onClick });
    expect(wrapper.find('button')).toHaveLength(1);
    expect(wrapper.find('a')).toHaveLength(0);
  });

  it('should render a <a> when onClick is not specified', () => {
    const href = 'https://www.lendi.com.au/';
    render({ href });
    expect(wrapper.find('button')).toHaveLength(0);
    expect(wrapper.find('a')).toHaveLength(1);
  });

  it('should render default styles when no size is specified', () => {
    const href = '#abc';
    render({ href });
    expect(wrapper.find(Link)).toHaveStyleRule('font-size', undefined);
  });

  it('should render default styles when no size or href is specified', () => {
    render({});
    expect(wrapper.find(Link)).toHaveStyleRule('font-size', '1em');
  });

  it('should render default styles for color when no color is specified', () => {
    render({ size: 'lg' });
    expect(wrapper.find(Link)).toHaveStyleRule('color', getColor('primary.500')({ theme }));
  });

  sizes.forEach((size, index) => {
    it(`should render styles for size ${size}`, () => {
      render({ size });
      expect(wrapper.find(Link)).toMatchSnapshot();
    });
  });

  colors.forEach((color) => {
    it(`should render styles for color ${color}`, () => {
      render({ size: 'lg', color });
      expect(wrapper.find(Link)).toHaveStyleRule('color', getColor(color)({ theme }));
    });
  });

  it('should render styles for margin', () => {
    render({ size: 'lg', m: 'md' });
    expect(wrapper.find(Link)).toHaveStyleRule('margin', `${deriveSize(1.5)}`);
  });
});
