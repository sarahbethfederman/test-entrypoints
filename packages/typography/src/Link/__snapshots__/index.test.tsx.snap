// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render styles for size lg 1`] = `
.c0 {
  font-size: 1em;
  border: none;
  padding: 0;
  font-size: calc(1.125 * var(--lendi-ui-size));
  line-height: calc(24 / 18);
  font-family: 'Open Sans',sans-serif;
  font-weight: bold;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: currentColor;
  background-color: inherit;
}

.c0:hover,
.c0:focus,
.c0:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<Component
  size="lg"
>
  <styled.button
    size="lg"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  font-size: 1em;
  border: none;
  padding: 0;
  ",
              [Function],
              " ",
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background-color: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "button",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="lg"
    >
      <button
        className="c0"
        size="lg"
      />
    </StyledComponent>
  </styled.button>
</Component>
`;

exports[`Link should render styles for size md 1`] = `
.c0 {
  font-size: 1em;
  border: none;
  padding: 0;
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
  font-weight: bold;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: currentColor;
  background-color: inherit;
}

.c0:hover,
.c0:focus,
.c0:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<Component
  size="md"
>
  <styled.button
    size="md"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  font-size: 1em;
  border: none;
  padding: 0;
  ",
              [Function],
              " ",
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background-color: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "button",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="md"
    >
      <button
        className="c0"
        size="md"
      />
    </StyledComponent>
  </styled.button>
</Component>
`;

exports[`Link should render styles for size sm 1`] = `
.c0 {
  font-size: 1em;
  border: none;
  padding: 0;
  font-size: calc(0.875 * var(--lendi-ui-size));
  line-height: calc(20 / 14);
  font-family: 'Open Sans',sans-serif;
  font-weight: bold;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: currentColor;
  background-color: inherit;
}

.c0:hover,
.c0:focus,
.c0:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<Component
  size="sm"
>
  <styled.button
    size="sm"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  font-size: 1em;
  border: none;
  padding: 0;
  ",
              [Function],
              " ",
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
  ",
              [Function],
              "
",
              ";
  background-color: inherit;
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "button",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="sm"
    >
      <button
        className="c0"
        size="sm"
      />
    </StyledComponent>
  </styled.button>
</Component>
`;
